[Skia] Stub out BackingStore implementation

donny-dont opened the pull request:
#### de89f2977a314e5ddbe49384e35b953061d17f29
<pre>
[Skia] Stub out BackingStore implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=270537">https://bugs.webkit.org/show_bug.cgi?id=270537</a>

Reviewed by Adrian Perez de Castro.

Add a stub of BackingStore when `USE(SKIA)` is turned `ON`. Actual
implementation to come later.

* Source/WebKit/Platform/WC.cmake:
* Source/WebKit/UIProcess/BackingStore.h:
* Source/WebKit/UIProcess/skia/BackingStoreSkia.cpp: Copied from Source\WebKit\UIProcess\BackingStore.h.

Canonical link: <a href="https://commits.webkit.org/275752@main">https://commits.webkit.org/275752@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/560f8e75fafc93e1de6bb988ee30d9f4cc66fe5a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42590 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21612 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44992 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45198 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38713 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25277 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18977 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35260 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43164 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18682 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36660 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16201 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16321 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/647 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38830 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38041 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46719 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17409 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14355 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41977 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19028 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40595 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9538 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19207 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18673 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/UIProcess/BackingStore.h:
```diff
@@ -41,6 +41,8 @@ struct UpdateInfo;
 #if USE(CAIRO)
 using PlatformPaintContextPtr = cairo_t*;
+#elif USE(SKIA)
+using PlatformPaintContextPtr = void*;
```
Any reason for the `void*` instead of `SkCanvas*`?

donny-dont commented on Source/WebKit/UIProcess/skia/BackingStoreSkia.cpp:
```diff
@@ -0,0 +1,64 @@
+BackingStore::BackingStore(const WebCore::IntSize& size, float deviceScaleFactor)
+    : m_size(size)
+    , m_deviceScaleFactor(deviceScaleFactor)
+    , m_scrolledHysteresis([this](PAL::HysteresisState state) { if (state == PAL::HysteresisState::Stopped) m_scrollSurface = nullptr; }, s_scrollHysteresisDuration)
```
@aperezdc the style checker doesn't like that which I don't think its ever going to like.

aperezdc replied:
Fair enough 😄

donny-dont committed:
[Skia] Stub out BackingStore implementation
https://bugs.webkit.org/show_bug.cgi?id=270537

Reviewed by Adrian Perez de Castro.

Add a stub of BackingStore when `USE(SKIA)` is turned `ON`. Actual
implementation to come later.

* Source/WebKit/Platform/WC.cmake:
* Source/WebKit/UIProcess/BackingStore.h:
* Source/WebKit/UIProcess/skia/BackingStoreSkia.cpp: Copied from Source\WebKit\UIProcess\BackingStore.h.

Canonical link: https://commits.webkit.org/275752@main
